pkg/machine/qemu/machine.go

```diff
@@ -599,27 +599,42 @@ func (v *MachineVM) Start(name string, opts machine.StartOptions) error {
 			_ = v.writeConfig()
 		}
 	}
-	if len(v.Mounts) > 0 {
-		connected := false
-		backoff = 500 * time.Millisecond
-		for i := 0; i < maxBackoffs; i++ {
-			if i > 0 {
-				time.Sleep(backoff)
-				backoff *= 2
-			}
-			state, err := v.State(true)
-			if err != nil {
-				return err
-			}
-			if state == machine.Running && v.isListening() {
-				connected = true
-				break
-			}
+	if len(v.Mounts) == 0 {
+		v.waitAPIAndPrintInfo(forwardState, forwardSock, opts.NoInfo)
+		return nil
+	}
+
+	connected := false
+	backoff = 500 * time.Millisecond
+	for i := 0; i < maxBackoffs; i++ {
+		if i > 0 {
+			time.Sleep(backoff)
+			backoff *= 2
+		}
+		state, err := v.State(true)
+		if err != nil {
+			return err
 		}
-		if !connected {
-			return fmt.Errorf("machine did not transition into running state")
+		if state == machine.Running && v.isListening() {
+			// Also make sure that SSH is up and running.  The
+			// ready service's dependencies don't fully make sure
+			// that clients can SSH into the machine immediately
+			// after boot.
+			//
+			// CoreOS users have reported the same observation but
+			// the underlying source of the issue remains unknown.
+			if err := v.SSH(name, machine.SSHOptions{Args: []string{"true"}}); err != nil {
+				logrus.Debugf("SSH readiness check for machine failed: %v", err)
+				continue
+			}
+			connected = true
+			break
 		}
 	}
+	if !connected {
+		return fmt.Errorf("machine did not transition into running state")
+	}
```

Comment:
Can we restructure this to actually include the information which part failed?
Basically if a users reports this we do not know if the machine is not running, not listening or ssh failed. IMO each case should have their own error in order to debug this and get data from users so we know were it fails.

Reply:
updated